File: models/HDU/Hangzhou Dianzi University/Model/archive/Model_Inference.py
import random
import numpy as np

Material = "Material A"

#%% Load Dataset

def load_dataset(in_file1="./Testing/"+Material+"/B_Field.csv", 
                 in_file2="./Testing/"+Material+"/Frequency.csv", 
                 in_file3="./Testing/"+Material+"/Temperature.csv"):
    
    data_B = np.genfromtxt(in_file1, delimiter=',') # N by 1024, in T
    data_F = np.genfromtxt(in_file2, delimiter=',') # N by 1, in Hz
    data_T = np.genfromtxt(in_file3, delimiter=',') # N by 1, in C

    return data_B, data_F, data_T

#%% Calculate Core Loss
def core_loss(data_B, data_F, data_T):
    
    #================ Wrap your model or algorithm here=======================#
    
    # Here's just an example:
    
    length = len(data_F)
    data_P = np.random.rand(length,1)
    
    #=========================================================================#
    
    with open("./Testing/Volumetric_Loss_"+Material+".csv", "w") as file:
        np.savetxt(file, data_P)
        file.close()

    print('Model inference is finished!')
    
    return

#%% Main Function for Model Inference

def main():
    
    # Reproducibility
    random.seed(1)
    np.random.seed(1)

    data_B, data_F, data_T = load_dataset()
    
    core_loss(data_B, data_F, data_T)
    
if __name__ == "__main__":
    main()
    
# End

File: finaltest/TeamName/Model/Model_Inference.m
clear all; close all; clc;

rng(1); % Reproducibility

global Material
Material = 'Material A';

main();

%% Run Main
function main()

    [data_B, data_F, data_T] = load_dataset();
    core_loss(data_B, data_F, data_T);

end

%% Load Dataset
function [data_B, data_F, data_T] = load_dataset()

    global Material
    
    data_B = readmatrix(['./Testing/', Material, '/B_Field.csv']); % N by 1024, in T
    data_F = readmatrix(['./Testing/', Material, '/Frequency.csv']); % N by 1, in Hz
    data_T = readmatrix(['./Testing/', Material, '/Temperature.csv']); % N by 1, in C

end

%% Calculate Core Loss
function core_loss(data_B, data_F, data_T)

    global Material

    %================ Wrap your model or algorithm here===================%
    
    % Here's just an example:
    
    len = length(data_F);
    data_P = rand(len,1);
    
    %=====================================================================%
    
    writematrix(data_P, ['./Testing/Volumetric_Loss_', Material, '.csv']);
    
    fprintf('Model inference is finished! \n');

end

%% End